src/ol/style/Icon.js

```diff
@@ -242,7 +242,7 @@ class Icon extends ImageStyle {
         this.updateScaleFromWidthAndHeight(this.width_, this.height_);
       };
       if (image.width > 0) {
-        this.updateScaleFromWidthAndHeight(this.width_, this.height_);
+        setScale();
```

Comment:
Should be `this.setScale()`, and the fact that this went unnoticed suggests that there is a test missing.

Reply:
No, it is the confusing named `setScale` local function defined immediately above, which is called either synchronously or asynchronously depending on whether the image is loaded.  If a function has been defined it should be used in both places - alternatively as it is only one line the `this.updateScaleFromWidthAndHeight(this.width_, this.height_)` could be repeated inside an anonymous callback for the load handler.